modules/terraform/validate.go

```diff
@@ -1,10 +1,29 @@
 package terraform
 
 import (
+	go_test "testing"
+
 	"github.com/gruntwork-io/terratest/modules/testing"
 	"github.com/stretchr/testify/require"
 )
 
+// Automatically finds all folders specified in RootDir that contain .tf files and runs InitAndValidate in all of them
+// Excludes any folders specified in the ExcludeDirs slice
+func ValidateAllTerraformModules(t *go_test.T, opts ValidationOptions) error {
+	dirsToValidate, readErr := ReadModuleAndExampleSubDirs(opts)
+	if readErr != nil {
+		return readErr
+	}
```

Comment:
Don't return error and use `require.NoError`. Technically, you should introduce a `ValidateAllTerraformModulesE` (note the `E` suffix) function that returns errors, but that will be hard to do with the subtest design of this function, so we should just introduce the non-E flavor only here.

Reply:
Heh, was going to say the same thing. Perhaps the `E` version of this method is a bit different and returns a list of test cases to run? And then the non `E` version loops over those test cases and calls `t.Run` on each one?

Reply:
Made the changes Yori propsoed in 2799cb1. Still thinking through Jim's suggestion.